### DISCUSSION TITLE
How to use another atom within read-only atom

### DISCUSSION THREAD

**Discussion by BilalAtique** [pmndrs/jotai#2092] [Q&A]
I am using atom for handling loading.
export const loadingAtom = atom(false)

I want to use this atom inside other atom. 
```js
export const compSetAtom = atom((get,) => {
  const data = get(dataAtom);
  const ticker = get(tickerAtom);
  const selectedPreset = get(selectedPresetAtom);
  const selectedOptions = get(selectedOptionsAtom);
  const presets = get(presetAtom)
  return generate(
    data,
    ticker,
    selectedPreset,
    selectedOptions,
    presets
  );
});
```
generate is a function that takes 4 to 6 seconds. I want to set the loading to true before running this function and when it is completed, I want to make it false again. 
Is it possible? I am new to Jotai and any help will be greatly appreciated.

**Comment by dai-shi** [comment]
Is `generate` an async function?

**Reply by BilalAtique** [reply]
No, generate is not an async function.

**Reply by dai-shi** [reply]
If a sync function takes several seconds, we can't do anything. It's not just Jotai limitation, but JS in general.
Would you consider moving the function to web workers?

**Reply by BilalAtique** [reply]
Yes, If it does not increase function time. Can you please guide how to leverage web workers here?

**Reply by dai-shi** [reply]
There's no magic with web workers. If you learn and move some tasks to web workers, your generate function will be async, and Jotai works with Suspense out of the box.

**Reply by BilalAtique** [reply]
I guess it will be a silly question. But can you tell why this code gives **_set is not a function_** error
```js
const compSetAtom = atom((get,) => {
  const cleanedData = get(cleanedDataAtom);
  const securityTicker = get(securityTickerAtom);
  const selectedPreset = get(selectedPresetAtom);
  const selectedOptions = get(selectedOptionsAtom);
  const presets = get(presetAtom)
  loadingAtom.write(true)
  const result =  genCompSet(
    cleanedData,
    securityTicker,
    selectedPreset,
    selectedOptions,
    presets
    );
    loadingAtom.write(false)
  return result
});
```

**Reply by dai-shi** [reply]
atom.write() is not something you explicitly invoke. In general, you can't do any side effects in your atom read function.
